Examples/Detectors/TGeoDetector/src/TGeoDetectorOptions.cpp

```diff
@@ -0,0 +1,297 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2018-2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "ActsExamples/TGeoDetector/TGeoDetectorOptions.hpp"
+
+#include "Acts/Definitions/Units.hpp"
+#include "ActsExamples/Utilities/Options.hpp"
+
+#include <cstdlib>
+#include <vector>
+
+#include <boost/program_options.hpp>
+
+void ActsExamples::Options::addTGeoGeometryOptions(Description& desc) {
+  using boost::program_options::value;
+
+  // due to the way program options handles options that can occur multiple
+  // times, all options of a logical block must always be present.
+  //
+  // each detector volume configuration is one logical block which can
+  // be repeated as many times as there are usable detector volumes.
+  //
+  // required per-volume options:
+  //
+  //   --geo-detector-volume InnerPixels
+  //   --geo-tgeo-sfbin-r-tolerance 5:5
+  //   --geo-tgeo-sfbin-phi-tolerance 0.025:0.025
+  //   --geo-tgeo-sfbin-z-tolerance 5:5
+  //   --geo-tgeo-nlayers 0  # boolean switch whether there are negative layers
+  //   --geo-tgeo-clayers 1  # boolean switch whether there are central layers
+  //   --geo-tgeo-players 0  # boolean switch whether there are positive layers
+  //
+  // within each volume there can be negative/central/positive layers depending
+  // on the which `--geo-tgeo-{n,c,p}layers` flags are set to true. if any of
+  // them are set, they must be followed by the corresponding layer option. if
+  // the `*layers` option is false, **no** further options **must** be set.
+  //
+  // examples: negative and central layers, but not positive layers
+  //
+  //   --geo-tgeo-nlayers 1
```

Comment:
As discussed, this could become a `boolean_switch` then.

Reply:
Yes, it should, but there does not seem to be support for a vector of boolean switches. It is implemented as a `std::vector<bool>`, but the option still needs an argument. Anything bool-like should work, e.g. `true`, `false`, etc. but I have not tested in in detail.